src/index.ts

```diff
@@ -287,18 +304,27 @@ class MagicNixCacheAction {
     const pid = parseInt(await fs.readFile(pidFile, { encoding: "ascii" }));
     actionsCore.debug(`found daemon pid: ${pid}`);
     if (!pid) {
-      throw new Error("magic-nix-cache did not start successfully");
+      const msg = "magic-nix-cache did not start successfully";
+      this.failInStrictMode(msg);
+      throw new Error(msg);
     }
 
     const log = tailLog(this.daemonDir);
 
     try {
       actionsCore.debug(`about to post to localhost`);
       const hostAndPort = inputs.getString("listen");
-      const res: Response = await this.client
-        .post(`http://${hostAndPort}/api/workflow-finish`)
-        .json();
-      actionsCore.debug(`back from post: ${res}`);
+      const res: Response<string> = await this.client.post(
+        `http://${hostAndPort}/api/workflow-finish`,
+      );
+
+      if (res.statusCode !== 200) {
+        this.failInStrictMode(
+          `Failed to trigger workflow finish hook; expected status 200 but got ${res.statusCode}`,
+        );
```

Comment:
I think it might be a good idea to include the response body in this message as well, so that you don't have to enable debug mode to see what the response was.

Reply:
I also meant that `res.body` should be available in the message passed to `this.failInStrictMode`. Otherwise, we'd only ever see the body (presumably telling us what went wrong) when rerunning the action and checking the "enable debug mode" checkbox, no?